src/pipe.js

```diff
@@ -30,5 +34,11 @@ var reverse = require('./reverse');
  *      squareThenDoubleThenTriple(5); //=> 150
  */
 module.exports = function pipe() {
-  return compose.apply(this, reverse(arguments));
+  return reduce(function(f, g) {
+    return curryN(f.length + g.length - 1, function() {
+      return g.apply(this,
+                     prepend(f.apply(this, take(f.length, arguments)),
+                             drop(f.length, arguments)));
```

Comment:
Nice. I didn't like this idea at the time I wrote the code because `fArgs` is all the arguments for `f` whereas `gArgs` is all *but one* of the arguments for `g`. There's a lack of symmetry.

I can look at it another way, though. We're partitioning `arguments` into a list of arguments for `f` and a list of arguments for `g`. We don't mean to imply that these are *all* the arguments in either case (and the code makes it clear that this is not so).